spec/lib/vanagon/project_spec.rb

```diff
@@ -413,6 +414,50 @@ class Project
     end
   end
 
+  describe '#save_manifest_json' do
+    include FakeFS::SpecHelpers
+    let(:platform_name) { 'el-7-x86_64' }
+    let(:platform) { Vanagon::Platform.new(platform_name) }
+    before(:each) do
+      class Vanagon
+        class Project
+          BUILD_TIME = '2018-07-10T13:34:25-07:00'
+          VANAGON_VERSION = '0.0.0-rspec'
+        end
+      end
+
+      @proj = Vanagon::Project.new('test-project', platform)
+    end
+
+    it 'should generate a file with the expected build metadata' do
+      correct_sample_metadata = {
+        'packaging_type' => { 'vanagon' => '0.0.0-rspec' },
+        'version' => '123abcde',
+        'components' => { 'test-component-10' => { 'version' => '1.2.3' } },
+        'build_time' => '2018-07-10T13:34:25-07:00',
+      }
+      bad_sample_metadata = {
+        'BAD KEY' => 'BAD VALUE'
+      }
+      comp1 = Vanagon::Component.new('test-component-10', {}, {})
+      comp1.version = '1.2.3'
+      @proj.components << comp1
+      @proj.version = '123abcde'
+      FakeFS do
```

Comment:
What is `FakeFS` / what's it for?

Reply:
FakeFS is an easy way to keep tests from leaving random files on the filesystem. It creates a temporary filesystem that only exists while it is in context.

https://rubygems.org/gems/fakefs

Reply:
ooh handy!